compiler/rustc_error_codes/src/error_codes/E0212.md

```diff
@@ -0,0 +1,52 @@
+An associated type cannot be extracted from
+a higher-ranked trait bound in the context.
+
+Erroneous code example:
+
+```compile_fail,E0212
+pub trait Foo<T> {
+    type A;
+
+    fn get(&self, t: T) -> Self::A;
+}
+
+fn foo2<I : for<'x> Foo<&'x isize>>(
+    field: I::A) {} // error!
+
+
+struct SomeStruct<I: for<'x> Foo<&'x isize>> {
+    x: I::A // error!
+}
+
+trait SomeTrait<I : for<'x> Foo<&'x isize>> {
+    fn some_method(&self, arg: I::A); // error!
+}
+```
+
+In this example, we have to instantiate `'x`, and
+we don't know what region to instantiate it with.
+To fix this, spell out the precise regions involved.
```

Comment:
I'm not sure exactly the difference between 'region' and 'lifetime', which means it will be even less clear to anyone reading the error for the first time.
```suggestion
we don't know what lifetime to instantiate it with.
To fix this, spell out the precise lifetimes involved.
```

Reply:
Me too. According to the [appendix/glossary](https://rustc-dev-guide.rust-lang.org/appendix/glossary.html), it seems simply mean lifetime.

Reply:
I applied suggestion. Would it be better to rebase this commit or be left it as it is?